statsmodels/tsa/statespace/mlemodel.py

```diff
@@ -332,7 +340,49 @@ def tolerance(self):
     def tolerance(self, value):
         self.ssm.tolerance = value
 
-    def fit(self, start_params=None, transformed=True,
+    def _validate_can_fix_params(self, param_names):
+        for param_name in param_names:
+            if param_name not in self.param_names:
+                raise ValueError('Invalid parameter name passed: "%s".'
+                                 % param_name)
+
+    @contextlib.contextmanager
+    def fixed_params(self, params):
+        from collections import OrderedDict
+
+        # Initialization (this is done here rather than in the constructor
+        # because param_names may not be available at that point)
+        if self._fixed_params is None:
+            self._fixed_params = {}
+            self._params_index = OrderedDict(
+                zip(self.param_names, np.arange(self.k_params)))
+
+        # Cache the current fixed parameters
+        old_fixed_params = self._fixed_params.copy()
+        old_has_fixed_params = self._has_fixed_params
+        old_fixed_params_index = self._fixed_params_index
+        old_free_params_index = self._free_params_index
+
+        # Validate parameter names and values
+        self._validate_can_fix_params(set(params.keys()))
+
+        # Set the new fixed parameters
+        self._fixed_params.update(params)
+        self._has_fixed_params = True
+        self._fixed_params_index = [self._params_index[key]
+                                    for key in params.keys()]
+        self._free_params_index = list(
+            set(np.arange(self.k_params)).difference(self._fixed_params_index))
+
+        yield
```

Comment:
should this have a try/finally so the reset below gets run?

Reply:
Yes! Thanks!